### PR TITLE
Update configuration option names

### PR DESCRIPTION
**Pull request by eoranged**

### CODE REVIEW

File: rq_dashboard/cli.py [hunk 1]
```diff
@@ -30,7 +31,34 @@ def basic_http_auth(*args, **kwargs):
...
+    'REDIS_PASSWORD': 'RQ_DASHBOARD_REDIS_PASSWORD',
+    'REDIS_DB': 'RQ_DASHBOARD_REDIS_DB',
+    'REDIS_SENTINELS': 'RQ_DASHBOARD_REDIS_SENTINELS',
+    'REDIS_MASTER_NAME': 'RQ_DASHBOARD_REDIS_MASTER_NAME',
+    'RQ_POLL_INTERVAL': 'RQ_DASHBOARD_POLL_INTERVAL',
+    'WEB_BACKGROUND': 'RQ_DASHBOARD_WEB_BACKGROUND',
+    'DELETE_JOBS': 'RQ_DASHBOARD_DELETE_JOBS',
+}
+
+warning_template = "Configuration option {old_name} is depricated and will be removed in future versions. "\
```
Comment (homeworkprod) [inline rq_dashboard/cli.py:47]: Typo; should read "deprecated".

File: README.rst [hunk 2]
```diff
@@ -49,29 +49,32 @@ Run the dashboard standalone, like this:
...
+      --password TEXT                 HTTP Basic Auth password
+      -c, --config TEXT               Configuration file (Python module on search
+                                      path)
+      -H, --redis-host TEXT           IP address or hostname of Redis server
+      -P, --redis-port INTEGER        Port of Redis server
+      --redis-password TEXT           Password for Redis server
+      -D, --redis-database INTEGER    Database of Redis server
+      -u, --redis-url TEXT            Redis URL connection (overrides other
+                                      individual settings)
+      --redis-sentinels TEXT          List of redis sentinels. Each should be
```
Comment (homeworkprod) [inline README.rst:67]: Should be "Redis".

File: README.rst [hunk 3]
```diff
@@ -49,29 +49,32 @@ Run the dashboard standalone, like this:
...
+                                      formatted: <host>:<port>
+      --redis-master-name TEXT        Name of redis master. Only needed when using
```
Comment (homeworkprod) [inline README.rst:69]: Should be "Redis".

File: README.rst [hunk 4]
```diff
@@ -49,29 +49,32 @@ Run the dashboard standalone, like this:
...
+                                      sentinels
+      --poll-interval, --interval INTEGER
+                                      Refresh interval in ms
```
Comment (homeworkprod) [inline README.rst:72]: Consider "milliseconds" to avoid confusion and to make it easier to grok.

File: rq_dashboard/cli.py [hunk 5]
```diff
@@ -30,7 +31,34 @@ def basic_http_auth(*args, **kwargs):
...
+
+
+def upgrade_config(app):
+    """
+    Updates old configuration options with new ones throwing warnings to those who haven't upgraded yet.
+    """
+    for old_name, new_name in LEGACY_CONFIG_OPTIONS.items():
+        if old_name in app.config:
+            warnings.warn(warning_template.format(old_name=old_name, new_name=new_name), DeprecationWarning)
+            app.config[new_name] = app.config[old_name]
```
Comment (homeworkprod) [inline rq_dashboard/cli.py:58]: I wonder if the old names should be removed from the app configuration object. Not sure about the consequences, though.

**Review by homeworkprod — APPROVED** [review]
This looks like a very reasonable approach to me.

I'm aware that a few of the comments I made are about things that haven't really been changed in this PR/have been this way before, but since I was around … :)

**Comment by homeworkprod** [comment]
All in all: Thank you for cleaning up with a broader perspective (rather than just targeting e.g. `WEB_BACKGROUND`). Much appreciated :)

File: rq_dashboard/cli.py (hunk 5 above)
Comment (eoranged) [reply rq_dashboard/cli.py:58]: Old names potentially can be used by other apps to perform their own configuration. That's why we can't remove them right away

File: rq_dashboard/cli.py (hunk 1 above)
Comment (eoranged) [reply rq_dashboard/cli.py:47]: I left typos as is for now. Feel free to submit a p/r with these corrections.
Anyway I'm planning big redesign and hit them all with a hammer during It anyways

**Pull request merged**